albumentations/augmentations/transforms.py

```diff
@@ -605,7 +606,104 @@ def get_transform_init_args_names(self):
+        return ["image"]
+
+    def get_transform_init_args_names(self):
+        return "sl", "sh", "r1", "num_holes", "fill_value"
+
+    def apply(self, img, holes=(), **kwargs):
+        for x, y, w, h in holes:
+            if img.size()[0] == 3:
+                img[0, x:x + w, y:y + h] = self.fill_value[0]
+                img[1, x:x + w, y:y + h] = self.fill_value[1]
+                img[2, x:x + w, y:y + h] = self.fill_value[2]
+            else:
+                img[0, x:x + w, y:y + h] = self.fill_value[0]
+        return img
+
+    def apply_to_bbox(self, bbox, holes=(), **params):
+        # TODO: This part could be tricky. If you accidentally erased an object then it should be removed.
+        raise NotImplementedError(
+            'This part could be tricky. If you accidentally erased an object then it should be removed.')
+
+    def apply_to_keypoint(self, keypoint, holes=(), **params):
+        x, y, angle, scale = keypoint[:4]
+        for hx, hy, hw, hh in holes:
+            if x >= hx and x < hw + hx and y >= hy and y < hh + hy:
+                return None
+            else:
+                return keypoint
+
+    def apply_to_keypoints(self, keypoints, **params):
+        kps = [self.apply_to_keypoint(tuple(keypoint[:4]), **params) + tuple(keypoint[4:]) for keypoint in keypoints]
+        while None in kps:
+            kps.remove(None)
+        return kps
+
+    def apply_to_mask(self, img, holes=(), **params):
+        for x, y, w, h in holes:
+            img[:, x:x + w, y:y + h] = 0
+        return img
+
+    def get_params_dependent_on_targets(self, params):
+        img = params["image"]
+        area = img.size()[1] * img.size()[2]
+        holes = []
+        num_holes = random.randint(1, self.num_holes - 1)
+
+        for h in range(num_holes):
+            target_ratio = random.uniform(self.sl, self.sh)
+            aspect_ratio = random.uniform(self.r1, 1 / self.r1)
+            target_area = target_ratio * area
+            flag = False
+            for attempt in range(10):
+                h = int(round(math.sqrt(target_area * aspect_ratio)))
+                w = int(round(math.sqrt(target_area / aspect_ratio)))
+                if w < img.size()[2] and h < img.size()[1]:
+                    x = random.randint(0, img.size()[1] - w)
+                    y = random.randint(0, img.size()[2] - h)
+                    holes.append((x, y, w, h))
+                    flag = True
+                    break
+            if not flag:
+                aspect_ratio = img.size()[2] / img.size()[1]
+                # Fallback to central crop
+                w = int(round(math.sqrt(target_area / aspect_ratio)))
+                h = int(round(math.sqrt(target_area * aspect_ratio)))
+                x = int(round((img.shape[0] - w) / 2))
+                y = int(round((img.shape[0] - h) / 2))
+                holes.append((x, y, w, h))
+        return {'holes': holes}
+
+
+class Cutout(DualTransform):
```

Comment:
Why did you change `Cutout` base class to `DualTransform`?

Reply:
Sorry, that's a mistake.